sync/src/synchronizer/block_process.rs

```diff
@@ -37,31 +37,30 @@ impl<'a> BlockProcess<'a> {
             block.hash(),
         );
         let snapshot = self.synchronizer.shared().snapshot();
+        let state = self.synchronizer.shared().state();
 
-        if self
-            .synchronizer
-            .shared()
-            .state()
-            .new_block_received(&block)
-            && self
+        if state.new_block_received(&block) {
+            if self
                 .synchronizer
                 .process_new_block(&snapshot, self.peer, block.clone())
                 .is_err()
-        {
-            info!(
-                "Ban peer {:?} for {} seconds, reason: it sent us an invalid block",
-                self.peer,
-                BAD_MESSAGE_BAN_TIME.as_secs()
-            );
-            self.synchronizer
-                .shared()
-                .state()
-                .insert_block_status(block.hash(), BlockStatus::BLOCK_INVALID);
-            self.nc.ban_peer(
-                self.peer,
-                BAD_MESSAGE_BAN_TIME,
-                String::from("send us an invalid block"),
-            );
+            {
+                info!(
+                    "Ban peer {:?} for {} seconds, reason: it sent us an invalid block",
+                    self.peer,
+                    BAD_MESSAGE_BAN_TIME.as_secs()
+                );
+                state.insert_block_status(block.hash(), BlockStatus::BLOCK_INVALID);
+                self.nc.ban_peer(
+                    self.peer,
+                    BAD_MESSAGE_BAN_TIME,
+                    String::from("send us an invalid block"),
+                );
+            }
+        } else if snapshot.contains_block_status(&block.hash(), BlockStatus::BLOCK_STORED) {
+            state
+                .peers()
+                .set_last_common_header(self.peer, block.header().clone());
```

Comment:
Is this likely to happen that `get_last_common_header(self.peer).number() > block.header.number()`, which means we replace with an old header.

Reply:
This peer may be reorganized, receive a smaller number header is fine.
We don't need do number comparison here.
We will fix it before we use it.
https://github.com/nervosnetwork/ckb/pull/1821/files#diff-86d69964b227ca1248fadb84da967a96R54-R55

Reply:
That's cool!